diagnostics-app/src/app/app.module.ts

```diff
@@ -9,15 +9,19 @@
 
 import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
-
 import { AppRoutingModule } from './app-routing.module';
-import { AppComponent } from './app.component';
 import { ServiceWorkerModule } from '@angular/service-worker';
 import { environment } from '../environments/environment';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 
+import { SharedModule } from './shared/shared.module';
+
+import { AppComponent } from './app.component';
+import { HeaderComponent } from './layout/header/header.component';
+import { CoreModule } from './core/core.module';
```

Comment:
Curious why you moved import around and if there is a best practice for import ordering?

Reply:
Re-arranged the imports. The best practice is to have all @angular imports on top, then the modules created inside the project and then the components.

Reply:
https://angular.io/guide/component-interaction#parent-and-children-communicate-using-a-service

According to this, it is much better to use `rxjs/Subject` rather than `EventEmitter`. I think this solves the purpose of abstracting the `emit` method to a function and provide only the `subscribe` object as public.

Reply:
Great. Let's do this in the other files as well.